tests/functional/test_new_resolver.py

```diff
@@ -1132,3 +1132,69 @@ def test_new_resolver_check_wheel_version_normalized(
         "simple"
     )
     assert_installed(script, simple="0.1.0+local.1")
+
+
+def test_new_resolver_does_reinstall_local_wheels(script):
```

Comment:
I definitely don't think that reinstalling wheels should be tested. IMO it's not something we should be guaranteeing, and therefore we shouldn't test it.

Do we have a use case for users expecting that installing a local wheel file forces a reinstall? I thought all the issues we'd seen revolved around *source* files/directories.

Reply:
Paul and I are discussing this further in #8711. Take your popcorn and go there. 🙃